UKHomeOffice/nmsw-ui
# NMSW-1073 Control when FAL1 can be re-uploaded

# Ticket



----

## AC

GIVEN I have a draft voyage
AND I have not uploaded any people details
THEN I am able to re-upload a FAL 1

----

## To test

SCENARIOS

**FAL 1 has been uploaded and user wants to change FAL 1**

- user uploads a FAL1
- user wants to change their FAL1
- user can reupload the FAL1 (they do not need to delete the draft)

To Test
- create a declaration and upload the FAL1
- > task list page should have a link to change FAL1
- click link
- > you should be taken to FAL1 upload page



**Passengers is YES, but no FAL6 yet and user wants to change FAL1**

- user uploads a FAL1
- user wants to change their FAL1
- user can reupload the FAL1 (they do not need to delete the draft)

To Test
- create a declaration and upload the FAL1
- select passengers and choose yes
- go to task list
- > it should have a link to change FAL 1
- click link
- > you should be taken to FAL1 upload page

**FAL 5 has been uploaded and user wants to change FAL 1**

- user uploads a FAL1
- user uploads a FAL5
- user wants to change their FAL1
- > user must delete draft and start again

To Test
- create a declaration and upload the FAL1
- upload a FAL5
- task list page should now have no link for FAL1
- go to FAL1 page by typing in browser /report-voyage/upload-general-declaration?report=<declarationId>
- you should be redirected to the task list page

**FAL 6 has been uploaded and user wants to change FAL 1**

- user uploads a FAL1
- user uploads a FAL6
- user wants to change their FAL1
- > user must delete draft and start again

To Test
- create a declaration and upload the FAL1
- select passengers YES
- upload a FAL6
- task list page should now have no link for FAL1
- go to FAL1 page by typing in browser /report-voyage/upload-general-declaration?report=<declarationId>
- you should be redirected to the task list page

----

## Notes
